src/posthog-core.ts

```diff
@@ -293,6 +293,7 @@ export class PostHog {
     webPerformance = new DeprecatedWebPerformanceObserver()
 
     _initialPageviewCaptured: boolean
+    _visibilityStateListener: (() => void) | null
```

Comment:
funnily enough i'm pretty sure we have at least two visibility state listeners now
i think maybe this is the third or fourth

so we could have a single listener that uses the event emitter which the other places react to

that _might_ be smaller (but is of course more complex ⚖️)

Reply:
I could keep calling this function on **any** visibility change because it's already guaranteeing that we only send the event once, so... maybe? I'll check

Reply:
This is actually just the second visibility listener, so I'm not too worried